src/components/WalletConnect/NanoContract/NewNanoContractTransactionModal.js

```diff
@@ -5,64 +5,52 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import React from 'react';
+import React, { useCallback, useEffect } from 'react';
 import {
   StyleSheet,
-  View,
   Text,
 } from 'react-native';
 import { t } from 'ttag';
 import { useNavigation } from '@react-navigation/native';
 import { useDispatch, useSelector } from 'react-redux';
 import { COLORS } from '../../../styles/themes';
-import { CircleInfoIcon } from '../../Icons/CircleInfo.icon';
 import { ModalBase } from '../../ModalBase';
-import SimpleButton from '../../SimpleButton';
-import { setNewNanoContractTransaction, walletWalletReject } from '../../../actions';
-import { WALLET_STATUS } from '../../../sagas/wallet';
+import { walletConnectReject } from '../../../actions';
 import { NANO_CONTRACT_INFO_URL } from '../../../constants';
+import { WarnDisclaimer } from '../WarnDisclaimer';
 
-export const NewNanoContractTransactionModal = () => {
+export const NewNanoContractTransactionModal = ({
+  onDismiss,
+  data,
+}) => {
+  const isRetrying = useSelector(({ walletConnect }) => (
+    walletConnect.newNanoContractTransaction.retrying
+  ));
   const navigation = useNavigation();
   const dispatch = useDispatch();
   const readMoreUrl = NANO_CONTRACT_INFO_URL;
 
-  const {
-    showModal,
-    ncTxRequest,
-  } = useSelector((state) => {
-    const {
-      walletStartState,
-      walletConnect: {
-        newNanoContractTransaction: {
-          showModal,
-          data,
-        }
-      },
-    } = state;
-    const isWalletReady = walletStartState === WALLET_STATUS.READY;
-
-    return {
-      showModal: showModal && isWalletReady,
-      ncTxRequest: data,
-    };
-  });
-
-  const onDismiss = () => {
-    dispatch(walletWalletReject());
-    dispatch(setNewNanoContractTransaction({ show: false, data: null }));
-  };
+  const onModalDismiss = useCallback(() => {
+    dispatch(walletConnectReject());
+    onDismiss();
+  }, [onDismiss]);
```

Comment:
`onDismiss` comes as a prop, so its reference might change, so we need to pass it as a dep